backend/api/Controllers/InspectionFindingsController.cs

```diff
@@ -0,0 +1,97 @@
+        /// </summary>
+        /// <remarks>
+        /// </remarks>
+        [HttpPost("add-findings")]
+        [Authorize(Roles = Role.Admin)]
+        [ProducesResponseType(typeof(InspectionFindings), StatusCodes.Status200OK)]
+        [ProducesResponseType(StatusCodes.Status400BadRequest)]
+        [ProducesResponseType(StatusCodes.Status409Conflict)]
+        [ProducesResponseType(StatusCodes.Status401Unauthorized)]
+        [ProducesResponseType(StatusCodes.Status403Forbidden)]
+        [ProducesResponseType(StatusCodes.Status500InternalServerError)]
+        public async Task<ActionResult<InspectionFindings>> AddFindings([FromBody] InspectionFindingsQuery inspectionFinding)
+        {
+            _logger.LogInformation("Updating inspection findings for inspection with isarStepId '{Id}'", inspectionFinding.IsarStepId);
+            try
+            {
+                var inspection = await _inspectionService.ReadByIsarStepId(inspectionFinding.IsarStepId);
+                if (inspection != null)
+                {
+                    inspection = await _inspectionService.AddFindings(inspectionFinding);
+                    return Ok(inspection?.InspectionFindings);
+                }
+
+            }
+            catch (Exception e)
+            {
+                const string Message = "Error while adding findings to inspection with IsarStepId '{Id}'";
+                _logger.LogError(e, "Message", inspectionFinding.IsarStepId);
+                return StatusCode(StatusCodes.Status500InternalServerError, Message);
+            }
+            return NotFound($"Could not find any inspection with the provided '{inspectionFinding.IsarStepId}'");
+        }
+
+        /// <summary>
+        /// Get the full inspection against an isarStepId
+        /// </summary>
+        /// <remarks>
+        /// </remarks>
+        [HttpGet]
+        [Authorize(Roles = Role.Admin)]
+        [Route("{id}")]
+        [ProducesResponseType(typeof(Inspection), StatusCodes.Status200OK)]
+        [ProducesResponseType(StatusCodes.Status400BadRequest)]
+        [ProducesResponseType(StatusCodes.Status409Conflict)]
+        [ProducesResponseType(StatusCodes.Status401Unauthorized)]
+        [ProducesResponseType(StatusCodes.Status403Forbidden)]
+        [ProducesResponseType(StatusCodes.Status500InternalServerError)]
+        public async Task<ActionResult<Inspection>> GetInspections([FromRoute] string id)
+        {
+            _logger.LogInformation("Get inspection by ID '{id}'", id);
+            try
+            {
+                var inspection = await _inspectionService.ReadByIsarStepId(id);
+                if (inspection != null)
+                {
+                    return Ok(inspection);
+                }
+
+            }
+            catch (Exception e)
+            {
+                const string Message = "Error while finding an inspection with inspection id '{id}'";
+                _logger.LogError(e, "Message", id);
```

Comment:
This will also just log "Message", look at other controllers to see how to log a message with a variable

Reply:
Yes my bad. Changed now.